symfony/client/pages.js

```diff
@@ -0,0 +1,33 @@
+const fs = require('fs');
+const path = require('path');
+
+// Traverse for plugin dirs
+function getPluginClientDirs(searchDir) {
+    return fs.readdirSync(searchDir).reduce((acc, localDir) => {
+        const pluginClientDir = path.join(searchDir, localDir, '/client');
+        if (fs.existsSync(pluginClientDir)) {
+            acc.push(pluginClientDir);
+        }
+        return acc;
+    }, []);
+}
+
+// Find Entry points
+function findEntryPoints(entryDir) {
+    const subDirs = fs.readdirSync(entryDir);
+    return subDirs instanceof Array ? subDirs.reduce((acc, localDir) => {
+        const entry = path.join(entryDir, localDir, '/index.js');
+        if (fs.existsSync(entry)) {
+            acc[localDir] = { entry };
+        }
+        return acc;
+    }, {}) : null;
+}
+
+const pages = getPluginClientDirs("../plugins/").map(entryDir => {
+    return findEntryPoints(entryDir);
+});
+
+// Write to pages.json
+fs.truncateSync('./pages.json', 0);
+fs.writeFileSync('./pages.json', JSON.stringify(Object.assign(...pages), null, 4), 'utf-8');
```

Comment:
Will that generate file in `{plugin-dir}/client/{page}/index.js` format ?

Reply:
This will search the plugin dir for entry points and generate [pages.json](https://github.com/orangehrm/orangehrm/blob/8a519298dcecf33a4b5a8814cbcfab08202fc192/symfony/client/pages.json) in below format. only have to run this when adding a new entry point. for the index.js format we will have to keep a util file in src and extend in each entry point.

```
{
    "jobCategory": {
        "entry": "../plugins/orangehrmAdminPlugin/client/jobCategory/index.js"
    }
}
```